src/consensus/consensus.rs

```diff
@@ -517,34 +621,8 @@ impl Consensus {
 
                 Ok(Resume::SignatureValidity(valid))
             }
-
-            Effect::Broadcast(gossip_msg) => {
-                match gossip_msg {
-                    SignedConsensusMsg::Proposal(proposal) => {
-                        debug!(
-                            "Broadcasting proposal gossip message: {:?} {:?} from {:?}",
-                            proposal.height, proposal.round, proposal.proposer
-                        );
-                        gossip_tx
-                            .send(GossipEvent::BroadcastSignedProposal(proposal))
-                            .await?;
-                    }
-                    SignedConsensusMsg::Vote(vote) => {
-                        debug!(
-                            "Broadcasting vote gossip message: {:?} {:?} {:?} from {:?}",
-                            vote.vote_type, vote.height, vote.round, vote.voter
-                        );
-                        gossip_tx
-                            .send(GossipEvent::BroadcastSignedVote(vote))
-                            .await?;
-                    }
-                }
-
-                Ok(Resume::Continue)
-            }
-
-            Effect::GetValue(height, round, timeout) => {
-                let timeout = timeouts.duration_for(timeout.step);
+            Effect::GetValue(height, round, timeout, _) => {
```

Comment:
There are a lot of these ignored params, are they all safe to ignore?

Reply:
Good point. I rewrote the code to use the extra param-- the code still does the same thing, the extra param is a more typesafe way of handling these resume values. It makes sure that the correct enum is used for the resume without us having to specify.